src/industry_gui.cpp

```diff
@@ -1437,11 +1437,12 @@ class IndustryDirectoryWindow : public Window {
 	 */
 	uint GetIndustryListWidth() const
 	{
-		uint width = 0;
-		for (const Industry *i : this->industries) {
-			width = std::max(width, GetStringBoundingBox(this->GetIndustryString(i)).width);
+		uint width = this->hscroll->GetCount();
+		auto [first, last] = this->vscroll->GetVisibleRangeIterators(this->industries);
+		for (auto it = first; it != last; ++it) {
+			width = std::max(width, GetStringBoundingBox(this->GetIndustryString(*it)).width);
 		}
-		return width + WidgetDimensions::scaled.framerect.Horizontal();
```

Comment:
Doesn't the scaled.framerect part still need to be added at some point?

Reply:
#12801 accounts for that correctly, it shouldn't have been part of the scrollbar's count.